uxarray/grid.py

```diff
@@ -401,22 +404,19 @@ def __from_ds__(self, dataset, multi_precision=False, precision=FLOAT_PRECISION_
         self._multi_precision = multi_precision
         self._precision = precision
 
-        # loops through the dataset
+        # convert coordinates to hmpy2.mpfr format
         if self._multi_precision:
-            for varName, varData in dataset.data_vars.items():
-                # check if the variable data is of float, integer, or string type and convert it based on the type
-                if np.issubdtype(varData.dtype, np.floating):
-                    convertedData = convert_to_multiprecision(
-                        varData.values, str_mode=False, precision=self._precision)
-                    self.ds[varName] = (varData.dims, convertedData)
-                elif np.issubdtype(varData.dtype, np.integer):
-                    convertedData = convert_to_multiprecision(
-                        varData.values, str_mode=False, precision=self._precision)
-                    self.ds[varName] = (varData.dims, convertedData)
-                elif np.issubdtype(varData.dtype, np.str):
-                    convertedData = convert_to_multiprecision(
-                        varData.values, str_mode=True, precision=self._precision)
-                    self.ds[varName] = (varData.dims, convertedData)
+            if "Mesh2_node_y" in self.ds and "Mesh2_node_x" in self.ds:
+                y_coords = self.ds["Mesh2_node_y"].values
+                x_coords = self.ds["Mesh2_node_x"].values
+
+                converted_y_coords = convert_to_multiprecision(
+                    y_coords, str_mode=False, precision=self._precision)
+                converted_x_coords = convert_to_multiprecision(
+                    x_coords, str_mode=False, precision=self._precision)
+
+                self.ds["Mesh2_node_y"] = converted_y_coords
+                self.ds["Mesh2_node_x"] = converted_x_coords
```

Comment:
What if the `self.islatlon == False`? Indicating that the input coordinates is 3D cartesian coordinates?